### PR TITLE
Adding notes support for Invoice and Subscription

### PR DESCRIPTION
**Pull request by bhelx**
Trying to catch .NET up to the other libraries with notes support.

#### Testing

You can run this and verify the notes have changed. If you run again you should be able to see they have persisted.

```csharp
var sub = Subscriptions.Get("2bf76b61f7874bb58c384d45759be62c");

Console.WriteLine(sub.CustomerNotes);
Console.WriteLine(sub.TermsAndConditions);
Console.WriteLine(sub.VatReverseChargeNotes);

Dictionary<string, string> notes = new Dictionary<string, string>();
notes.Add("CustomerNotes", "New Customer Notes");
notes.Add("TermsAndConditions", "New T and C");
notes.Add("VatReverseChargeNotes", "New VAT Notes");
sub.UpdateNotes(notes);

Console.WriteLine(sub.CustomerNotes);
Console.WriteLine(sub.TermsAndConditions);
Console.WriteLine(sub.VatReverseChargeNotes);
```

### CODE REVIEW

File: Library/Subscription.cs [hunk 1]
```diff
@@ -341,6 +345,29 @@ public void Postpone(DateTime nextRenewalDate)
                 ReadXml);
         }
 
+        public bool UpdateNotes(String customerNotes, String termsAndConditions, String vatReverseChargeNotes)
```
Comment (bhelx) [inline Library/Subscription.cs]: Should we be passing in some kind of hasmap here?

File: Library/Subscription.cs [hunk 2]
```diff
@@ -341,6 +345,29 @@ public void Postpone(DateTime nextRenewalDate)
...
+        {
+
+            if (customerNotes != null)
+                CustomerNotes = customerNotes;
```
Comment (bhelx) [inline Library/Subscription.cs]: Not sure if it's the best idea to set this on the client side subscription object. What would/should happen if the API call fails?

File: Library/Subscription.cs [hunk 3]
```diff
@@ -341,6 +345,29 @@ public void Postpone(DateTime nextRenewalDate)
...
+            if (customerNotes != null)
+                VatReverseChargeNotes = vatReverseChargeNotes;
+
+            Client.Instance.PerformRequest(Client.HttpRequestMethod.Put,
+                UrlPrefix + Uri.EscapeUriString(Uuid) + "/notes",
+                WriteSubscriptionNotesXml,
+                ReadXml);
+
+            // this method does not save the object
+            _saved = false;
```
Comment (bhelx) [inline Library/Subscription.cs]: Not sure if this is needed.

File: Library/Subscription.cs (hunk 1 above)
Comment (cbarton) [inline Library/Subscription.cs]: Ya, I like that idea

File: Library/Subscription.cs (hunk 2 above)
Comment (cbarton) [inline Library/Subscription.cs]: Maybe we can pass in the notes to the `WriteSubscriptionNotesXml` method and then if the save is successful assign them to the subscription

**Comment by bhelx** [comment]
:+1:

**Pull request merged**